src/google/protobuf/extension_set.h

```diff
@@ -419,13 +419,8 @@ class LIBPROTOBUF_EXPORT ExtensionSet {
   // are never forced to serialize deterministically.
   uint8* SerializeWithCachedSizesToArray(int start_field_number,
                                          int end_field_number,
-                                         uint8* target) const {
```

Comment:
We did a trick for ExtensionSet:
1. All methods are declared in extensoin_set.h, included by both lite runtime and full runtime.
2. Some methods are defined in extension_set.cc, included by both lite runtime and full runtime.
3. Some other methods (including all SerializeXXXToArray methods) are defined in extension_set_heavy.cc, included by full runtime only.

So when we build the lite runtime, some methods are actually undefined (though they are declared), but the compiler doesn't report any error because these methods are not actually called by anyone. These inline functions changed this though. They are calling other methods defined in extension_set_heavy.cc which is not part of lite runtime and different compilers end up with different behaviors:
1. gcc works probably because it finds these inline functions are not actually used in lite runtime so it ignores the entire thing (even these inline functions are calling undefined functions).
2. MSVC is not so nice and complains because it can't find the definition of the called functions.

My change fixes the problem by removing these inline functions.

The real problem I think is how we implemented ExtensionSet. We should have created two classes ExtensionSetLite and ExtensionSet, but it's probably not worthwhile to change it now.